### PR TITLE
Add HTTP CONNECT proxy support via ProxyOptions

### PR DESCRIPTION
**Pull request by onurmus**
## Description

Adds a `ProxyOptions` Pydantic model to `temporallib.client` and wires it into `Options`, allowing users to configure an _HTTP proxy_ through the standard `Options/env-var` pattern the library already uses for auth, encryption, and TLS.

### Why
The Temporal Rust core does not read `HTTPS_PROXY / HTTP_PROXY / GRPC_PROXY / NO_PROXY` environment variables. See the [issue here](https://github.com/grpc/grpc-rust/issues/2298). The only supported path is the explicit `http_connect_proxy_config` config field. Previously, `temporallib` did not forward this parameter, so users in proxy-only egress environments couldn't connect without bypassing the wrapper.

### Changes
* Add `ProxyOptions(BaseSettings) ` model with host, username, and password (SecretStr) fields; reads from `TEMPORAL_PROXY_HOST`, `TEMPORAL_PROXY_USERNAME`, `TEMPORAL_PROXY_PASSWORD`.
* Add unit tests covering both the forwarded-value and default-None cases.
* Bump version 1.8.7 → 1.9.0.

### Compatibility
Fully backward compatible. The parameter defaults to None, so existing callers are unaffected. No changes to auth, encryption, TLS, or the reconnect loop (the proxy set at connect time persists across reconnects).

## Engineering checklist
_Check only items that apply_

- [ ] Documentation updated
- [x] Have tested the workflow works as expected

## Test instructions
* Added new test cases for the change
* All 11 tests pass.

### CODE REVIEW

**Review by RenanGreca — APPROVED** [review]
LGTM, great work

File: temporallib/client/client.py [hunk 1]
```diff
@@ -156,6 +161,7 @@ async def connect(
         lazy: bool = False,
         runtime: Optional[Runtime] = None,
         keep_alive_config: Optional[KeepAliveConfig] = None,
+        http_connect_proxy_config: Optional[HttpConnectProxyConfig] = None,
```
Comment (RenanGreca) [inline temporallib/client/client.py]: Actually, wouldn't it make sense to add the proxy to the `Options` configuration? This way it can be read directly from the environment via Pydantic without needing extra code in the worker.

**Review by RenanGreca — CHANGES_REQUESTED** [review]
Please consider implementing the proxy configuration as a Pydantic validator in the `Options` class.

**Review by RenanGreca — APPROVED** [review]
Fantastic!

File: temporallib/client/client.py [hunk 2]
```diff
@@ -6,20 +6,34 @@
...
 from temporallib.auth import AuthHeaderProvider, AuthOptions
 from temporallib.encryption import EncryptionOptions, EncryptionPayloadCodec
 
 logging.basicConfig(level=logging.INFO)
 
 
+class ProxyOptions(BaseSettings):
+    host: Optional[str] = None
+    username: Optional[str] = None
+    password: Optional[SecretStr] = None
```
Comment (RenanGreca) [inline temporallib/client/client.py]: TIL about `SecretStr`, good to know.

**Review by stasiaaleks — APPROVED** [review]
LGTM!

**Comment by RenanGreca** [comment]
See [this comment](https://github.com/canonical/assessment-hub/pull/26#discussion_r3869749029)

File: temporallib/client/client.py [hunk 3]
```diff
@@ -28,6 +42,7 @@ class Options(BaseSettings):
     tls_root_cas: Optional[str] = None
     auth: Optional[AuthOptions] = None
     prometheus_port: Optional[str] = None
+    proxy: Optional[ProxyOptions] = None
```
Comment (nikita-jare) [inline temporallib/client/client.py]: Does `Options()` on its own pick this up from the env? I don't think it does.
pydantic-settings doesn't apply the child's `TEMPORAL_PROXY_`  prefix from the parent, so proxy stays `None` unless the caller passes `Options(proxy=ProxyOptions())`

File: temporallib/client/client.py [hunk 4]
```diff
@@ -99,6 +114,17 @@ async def _get_auth_headers(self, auth: AuthOptions) -> Mapping[str, str]:
...
         return await loop.run_in_executor(None, auth_header_provider.get_headers)
 
+    @classmethod
+    def _build_proxy_config(
+        self, proxy: ProxyOptions
+    ) -> Optional[HttpConnectProxyConfig]:
+        if not proxy or not proxy.host:
+            return None
+        basic_auth = None
+        if proxy.username and proxy.password:
```
Comment (nikita-jare) [inline temporallib/client/client.py]: non-blocking: if only the username or only the password is set, we drop both silently and fail later with a 407 from the proxy. Probably better to raise here.

File: temporallib/client/client.py (hunk 3 above)
Comment (nikita-jare) [reply temporallib/client/client.py]: Adding an env-var test would be good too, matching `test_connect_google_env_variables`.

**Review by nikita-jare — COMMENTED** [review]
Changes look good. Few small things below:
Can you also update the README with the new option & the TEMPORAL_PROXY_* vars.

**Review by nikita-jare — APPROVED** [review]
Nice! LGTM. Thank you @onurmus
